File: cakk-domain/src/main/java/com/cakk/domain/converter/DayOfWeekConverter.java
package com.cakk.domain.converter;

import java.util.stream.Stream;

import com.cakk.common.enums.DayOfWeek;
import com.cakk.common.enums.ReturnCode;
import com.cakk.common.exception.CakkException;

import jakarta.persistence.AttributeConverter;

public class DayOfWeekConverter implements AttributeConverter<DayOfWeek, Integer> {

	@Override
	public Integer convertToDatabaseColumn(DayOfWeek dayOfWeek) {
		if (dayOfWeek == null) {
			return null;
		}

		return dayOfWeek.getCode();
	}

	@Override
	public DayOfWeek convertToEntityAttribute(Integer code) {
		if (code == null) {
			return null;
		}

		return Stream.of(DayOfWeek.values())
			.filter(dayOfWeek -> dayOfWeek.getCode().equals(code))
			.findFirst()
			.orElseThrow(() -> new CakkException(ReturnCode.INTERNAL_SERVER_ERROR));
	}
}

File: cakk-infrastructure/persistence-mysql/src/main/java/com/cakk/infrastructure/persistence/converter/DayOfWeekConverter.kt
package com.cakk.infrastructure.persistence.converter

import com.cakk.common.enums.Days
import com.cakk.common.enums.ReturnCode
import com.cakk.common.exception.CakkException
import jakarta.persistence.AttributeConverter
import java.util.stream.Stream

class DayOfWeekConverter : AttributeConverter<Days?, Int?> {

	override fun convertToDatabaseColumn(days: Days?): Int? {
        return days?.code
    }

    override fun convertToEntityAttribute(code: Int?): Days? {
        return if (code == null) {
            null
        } else Stream.of(*Days.values())
                .filter { days: Days -> days.code == code }
                .findFirst()
                .orElseThrow { CakkException(ReturnCode.INTERNAL_SERVER_ERROR) }
    }
}

